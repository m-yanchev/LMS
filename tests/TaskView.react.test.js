import React from 'react';
import TaskView from '../main/components/views/TaskView/'
import renderer from 'react-test-renderer'

jest.mock('../client/components/common/Content/');

test('TaskView', () => {

    let component = renderer.create(
        <TaskView
            itemProps={
                {
                    isEditMode: false,
                    contentData: {
                        id: 1,
                        desc: 'Задание'
                    }
                }
            }
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <TaskView
            itemProps={
                {
                    isEditMode: false,
                    contentData: {
                        id: 1,
                        commonDesc: 'Общее задание',
                        desc: 'Задание',
                        src: 'источник',
                        srcNumb: 'номер в источнике',
                        solution: 'решение',
                        answer: 'ответ'
                    }
                }
            }
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});