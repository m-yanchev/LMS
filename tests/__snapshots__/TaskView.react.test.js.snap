// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView 1`] = `
<div
  className="task-view"
/>
`;

exports[`TaskView 2`] = `
<div
  className="task-view"
>
  <p
    className="common-desc"
  >
    Общее задание:
  </p>
  <div
    className="src"
  >
    <h5>
      Источник:
    </h5>
    <p>
      источник
    </p>
  </div>
  <div
    className="srcNumb"
  >
    <h5>
      Номер источника:
    </h5>
    <p>
      номер в источнике
    </p>
  </div>
  <div
    className="content-with-header content-with-header"
  >
    <h5
      className="content-with-header-header content-with-header-header"
    >
      Решение:
    </h5>
  </div>
  <div
    className="content-with-header content-with-header"
  >
    <h5
      className="content-with-header-header content-with-header-header"
    >
      Ответ:
    </h5>
  </div>
</div>
`;
